### ISSUE TITLE
[Application bug / Parameter change] OnBeforeGetEmailAttachmentsByEmailScenarios > var

### ISSUE THREAD

**Issue by fvet** [microsoft/ALAppExtensions#27230]
### Describe the request

The OnBeforeGetEmailAttachmentsByEmailScenarios in Email.Codeunit.al  allows adding filters to the Email Scenario Attachments before they are retrieved. The EmailScenarioAttachments parameter is however not set by var.

```
 /// <summary>
    /// Integration event that allows adding filters to the Email Scenario Attachments before they are retrieved.
    /// </summary>
    /// <param name="EmailScenarioAttachments">The record to add filters to.</param>
    [IntegrationEvent(false, false)]
    internal procedure OnBeforeGetEmailAttachmentsByEmailScenarios(EmailScenarioAttachments: Record "Email Scenario Attachments")
    begin
    end;
```

https://github.com/microsoft/BCApps/blob/f591cf2931ed2df68a8b8066129ae9ffb1b059e4/src/System%20Application/App/Email/src/Email/Email.Codeunit.al#L642

Expected code

```
 /// <summary>
    /// Integration event that allows adding filters to the Email Scenario Attachments before they are retrieved.
    /// </summary>
    /// <param name="EmailScenarioAttachments">The record to add filters to.</param>
    [IntegrationEvent(false, false)]
    internal procedure OnBeforeGetEmailAttachmentsByEmailScenarios(var EmailScenarioAttachments: Record "Email Scenario Attachments")
    begin
    end;
```

### Additional context

x
Internal work item: [AB#548197](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/548197)

**Comment by fvet** [comment]
@JesperSchulz 

### Additional Context

`GetEmailAttachmentsByEmailScenarios` seems to be used for building the 'Email Scenario Attach Setup' page content (1) and by 'AddAttachmentToMessage' for including the email scenario attachments to the email message (2).

We're especially interested in scenario (2), as we would like to add a 'language code' to the email scenario attachments, allowing us to specify a different email attachment **by language**.

If changing the `OnBeforeGetEmailAttachmentsByEmailScenarios` signature would be too difficult (since breaking by setting 'var' - although I don't see how this even can be useful without var access), our problem could get solved by introducing a new event as well on this location:

![Image](https://github.com/user-attachments/assets/29133fc4-7dde-4c0f-a2d0-ec76521bfc07)

**Comment by pri-kise** [comment]
This issue seems to be a duplicate of https://github.com/microsoft/BCApps/issues/3034 .

**Comment by JesperSchulz** [comment]
> This issue seems to be a duplicate of [microsoft/BCApps#3034](https://github.com/microsoft/BCApps/issues/3034) .

Indeed. The internal work item had been closed as duplicate, but GitHub was forgotten - not the first time 😟 But luckily we've got bright people like you monitoring what's going on out here and helping us get things right 🙏
